enumerables.rb

```diff
@@ -0,0 +1,134 @@
+module Enumerable
+  # My Each Method
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = self.class == Array ? self : to_a
+    counter = 0
+    while counter < arr.length
+      yield(arr[counter])
+      counter += 1
+    end
+    arr
+  end
```

Comment:
## Range and array self
- [ ] Test with hashes and ranges, then compare with original `each` Ruby method
- [ ] Fix your method to work with the tests. Follow this [sample](https://gist.github.com/bertil291utn/5a7e6a73351c38ceb9781a9a37b58cf6)

**Example:**

`test_file.rb`
```ruby
(0..9).my_each{
# or
Range.new(0, 9)

```

Reply:
Changed the structure of the code tested the code to ensure the above test examples are working perfectly.

Reply:
Refactor the code: Removed the outdated push symbols and replaced it with push built-in method.

Reply:
All test were tested and working properly